src/main/java/org/kuali/kfs/kew/engine/BlanketApproveEngine.java

```diff
@@ -0,0 +1,331 @@
+        }
+        return buffer.toString();
+    }
+
+    /**
+     * Invokes the blanket approval for the given node instance. This deactivates all pending approve or complete
+     * requests at the node and sends out notifications to the individuals who's requests were trumped by the blanket
+     * approve.
+     */
+    private void invokeBlanketApproval(ActionTaken actionTaken, RouteNodeInstance nodeInstance,
+            NotificationContext notifyContext) {
+        List actionRequests = getActionRequestService()
+                .findPendingRootRequestsByDocIdAtRouteNode(nodeInstance.getDocumentId(),
+                        nodeInstance.getRouteNodeInstanceId());
+        actionRequests = getActionRequestService().getRootRequests(actionRequests);
+        List<ActionRequest> requestsToNotify = new ArrayList<>();
+        for (Iterator iterator = actionRequests.iterator(); iterator.hasNext(); ) {
+            ActionRequest request = (ActionRequest) iterator.next();
+            if (request.isApproveOrCompleteRequest()) {
+                getActionRequestService().deactivateRequest(actionTaken, request);
+                requestsToNotify.add(request);
+            }
+            //KULRICE-12283: Added logic to deactivate acks or FYIs if a config option is provided. This will mainly
+            //be used when a document is moved.
+            if (request.isAcknowledgeRequest() && config.isDeactivateAcknowledgements()) {
+                getActionRequestService().deactivateRequest(actionTaken, request);
+            }
+            if (request.isFYIRequest() && config.isDeactivateFYIs()) {
+                getActionRequestService().deactivateRequest(actionTaken, request);
+            }
+        }
+        if (notifyContext != null && !requestsToNotify.isEmpty()) {
+            ActionRequestFactory arFactory =
+                    new ActionRequestFactory(RouteContext.getCurrentRouteContext().getDocument(), nodeInstance);
+            KimPrincipalRecipient delegatorRecipient = null;
+            if (actionTaken.getDelegatorPrincipal() != null) {
+                delegatorRecipient = new KimPrincipalRecipient(actionTaken.getDelegatorPrincipal());
+            }
+            List<ActionRequest> notificationRequests = arFactory
+                    .generateNotifications(requestsToNotify, notifyContext.getPrincipalTakingAction(),
+                            delegatorRecipient, notifyContext.getNotificationRequestCode(),
+                            notifyContext.getActionTakenCode());
+            getActionRequestService().activateRequests(notificationRequests);
+        }
+    }
+
+    private ActionRequestService getActionRequestService() {
+        return KEWServiceLocator.getActionRequestService();
+    }
+
+    private class ProcessEntry {
+
+        private RouteNodeInstance nodeInstance;
+        private int timesProcessed = 0;
+
+        ProcessEntry(RouteNodeInstance nodeInstance) {
```

Comment:
This was changed from public ProcessEntry to Process Entry. Is that desired?

Reply:
This functions visibility was pulled in from base code.  I only brought in the one line, which was KewApiConstants.ACTION_REQUEST_FYI_REQ.

When KualiCo brought this file into KFS, this function had the visibility as it is now.  I am not sure why or when KualiCo changed the visibility but I think we should go with it as is.

Also, since this is a private internal sub-class, the function was only ever visible from inside this class.